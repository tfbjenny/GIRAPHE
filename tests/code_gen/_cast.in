node a = node(1);
node b = node(1.2);
node c = node(true);
node d = node("Hello");
graph g = null;

/* int() */

print( int(23) );
print( int(a) );
print( int(b) );
g = a -- 2& b;
print( int( g@(a,b) ) );
print( int( (a -- 4.4& b)@(a,b) ) );

/* bool() */

print( bool(1>3) );
print( bool(1<3) );
print( bool(a) );
print( bool(b) );
print( bool(c) );
g = a -- (2<3)& b;
print( bool( g@(a,b) ) );
print( bool( (a -- 4.4& b)@(a,b) ) );

/* float() */

print( float(23) );
print( float(3.4) );
print( float(a) );
print( float(b) );
g = a -- 3.2& b;
print( float( g@(a,b) ) );
print( float( (a -- 6& b)@(a,b) ) );

/* string() */

print( string("Hello") );
print( string(d) );
print( string( (a -- "Edge"& b)@(a,b) ) );
