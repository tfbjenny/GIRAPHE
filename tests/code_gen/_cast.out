23
1
1
2
4
false
true
true
true
true
true
true
23.000000
3.400000
1.000000
1.200000
3.200000
6.000000
Hello
Hello
Edge
